src/main/java/graphql/execution/Execution.java

```diff
@@ -77,8 +77,14 @@ public CompletableFuture<ExecutionResult> execute(Document document, GraphQLSche
         List<VariableDefinition> variableDefinitions = operationDefinition.getVariableDefinitions();
 
         CoercedVariables coercedVariables;
+        NormalizedVariables normalizedVariableValues;
         try {
             coercedVariables = ValuesResolver.coerceVariableValues(graphQLSchema, variableDefinitions, inputVariables, executionInput.getGraphQLContext(), executionInput.getLocale());
+
+            normalizedVariableValues = ValuesResolver.getNormalizedVariableValues(graphQLSchema,
+                    variableDefinitions,
+                    inputVariables,
+                    executionInput.getGraphQLContext(), executionInput.getLocale());
```

Comment:
This is new -  we will coerce variables into rutime form AND Normalised literal form so that the ExecutionContext can have them AND the `QueryDirectives` can have them in the DFE